near-contract-standards/src/non_fungible_token/core/core_impl.rs

```diff
@@ -300,18 +300,23 @@ impl NonFungibleToken {
     ) -> Token {
         assert_eq!(env::predecessor_account_id(), self.owner_id, "Unauthorized");
 
-        self.internal_mint(token_id, token_owner_id, token_metadata)
+        self.internal_mint(token_id, token_owner_id, token_metadata, Some(self.owner_id.clone()))
     }
 
     /// Mint a new token without checking:
     /// * Whether the caller id is equal to the `owner_id`
+    /// * `refund_id` will transfer the left over balance after storage costs are calculated to the provided account.
+    ///   Typically the account will be the owner. If `None`, will not refund. This is useful for delaying refunding
+    ///   until multiple tokens have been minted.
+    ///
     pub fn internal_mint(
         &mut self,
         token_id: TokenId,
         token_owner_id: AccountId,
         token_metadata: Option<TokenMetadata>,
+        refund_id: Option<AccountId>,
     ) -> Token {
-        let initial_storage_usage = env::storage_usage();
+        let initial_storage_usage = if refund_id.is_some() { env::storage_usage() } else { 0 };
```

Comment:
Can you put this in the if block below? Doesn't really make sense to do this check twice when only used below if Some

Reply:
This was to avoid calling `env::storage_usage` if not refunding.  I just updated it to be a little more rusty, by using map to make this an option.